### PR TITLE
Added events using the same mechanism as static checkout pages

### PR DESCRIPTION
**Pull request by inventhouse**
Added some test events using the same mechanism we intend to use for REV-1074 static checkout pages

Note that I wrote the `sendRev1074Event` with an eye to being able to just drop it into the static pages when we add events there.

### CODE REVIEW

File: src/payment/cart/OrderSummary.jsx [hunk 1]
```diff
@@ -12,6 +13,12 @@ class OrderSummary extends React.Component {
       'edx.bi.ecommerce.payment_mfe.order_summary_rendered',
       getPerformanceProperties(),
     );
+    let eventData = {};
+    let renderTime = window.performance.getEntriesByName("Order Summary component rendered");
+    if (renderTime) {
+      eventData["summaryRenderTiming"] = renderTime[renderTime.length - 1].toJSON();
```
Comment (MatthewPiatetsky) [inline src/payment/cart/OrderSummary.jsx]: what is this?

File: src/index.jsx [hunk 2]
```diff
@@ -49,6 +49,8 @@ subscribe(APP_READY, () => {
     'edx.bi.ecommerce.payment_mfe.started_painting',
     getPerformanceProperties(),
   );
+  sendRev1074Event('payment_mfe.started_painting', {});
```
Comment (MatthewPiatetsky) [inline src/index.jsx]: you're adding all of these after sendTrackEvent but doesn't that function go to the same api?

File: src/payment/cart/OrderSummary.jsx (hunk 1 above)
Comment (inventhouse) [reply src/payment/cart/OrderSummary.jsx]: That gets the timing mark for when the Order Summary component finishes rendering (if available; taking the last one if multiple, which there shouldn't be, but `getEntriesByName` returns an array)

File: src/payment/index.js [hunk 3]
```diff
@@ -1,3 +1,33 @@
...
+/* eslint-disable prefer-template */
+/* eslint-disable dot-notation */
+export function sendRev1074Event(eventType, eventData) {
+  eventData['_export'] = 'false'; // Don't let these events be exported to partners
+  if (window.performance.timing) {
+    eventData.timing = window.performance.timing.toJSON();
+  }
+  const encodedEvent = [
+    'event_type=edx.experiment.rev1074.' + encodeURIComponent(eventType),
+    'page=' + encodeURIComponent(window.location.href),
```
Comment (MatthewPiatetsky) [inline src/payment/index.js]: why do we need to encode the different variables? is this for security or something

File: src/payment/cart/OrderSummary.jsx (hunk 1 above)
Comment (MatthewPiatetsky) [reply src/payment/cart/OrderSummary.jsx]: so what happens if renderTime is not defined? do we not have a time? Should the sendEvent be inside this if block then?

File: src/payment/index.js (hunk 3 above)
Comment (inventhouse) [reply src/payment/index.js]: As far as I could tell, the GET does _not_ encode them automatically, please let me know if there's a better way.  (Note: this is just encoding, not encrypting, no security implications around this)

File: src/payment/cart/OrderSummary.jsx (hunk 1 above)
Comment (inventhouse) [reply src/payment/cart/OrderSummary.jsx]: If we don't have a `renderTime`, oh well, but we should still send the event anyway so we can compare with the regular track event.

File: src/index.jsx (hunk 2 above)
Comment (inventhouse) [reply src/index.jsx]: No, I believe the regular track events are going to Segment.

File: src/payment/index.js (hunk 3 above)
Comment (MatthewPiatetsky) [reply src/payment/index.js]: why is the encoding necessary in that case? if not necessary we could save some characters on the static page

Comment (inventhouse) [reply src/payment/index.js]: Ooohhh, you mean _that_ specific instance; probably not?  (The "spec" is _very_ unclear about what needs to be encoded in different cases)  I'll try without encoding that one 👍

File: src/index.jsx (hunk 2 above)
Comment (inventhouse) [reply src/index.jsx]: Mainly they're going to different destinations; by adding "duplicate" events going to the API we plan to use for the Static Checkout Page test we can validate that mechanism and try out analyzing these events.

File: src/payment/index.js (hunk 3 above)
Comment (inventhouse) [reply src/payment/index.js]: Dropped the encode calls on things that don't really need it.

**Pull request merged**